SalesforceCommerceCloud/pwa-kit
# [V2] Implement dev server serveStaticFile

# Description



We recently added `serveStaticFile` to the runtime api, however we noticed that it only works on a remote environment, it throws a 404 error on local development. This is because that the remote server `serveStaticFile` implementation relies on a `/build` folder that may not exists on local file system.

The tricky part about this is that static files are inside `/app/static` and the files gets copied over to `/build/static` by webpack. We need to re-write the paths.

The solution is to implement `serveStaticFile` in the `DevServerMixin`. The implementation is based on the webpack dev middleware context and we can leverage the the source file path from the stats.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- implement serveStaticFile in dev server

# How to Test-Drive This PR

- `npm ci`
- cd packages/template-typescript-minimal
- npm start (make sure you don't have `/build`, remove the folder if you do)
- go to `http://localhost:3000/favicon.ico` and verify the file is served correctly

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
